modules/flocIdSystem.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * This module adds flocId to the User ID module
+ * The {@link module:modules/userId} module is required
+ * @module modules/flocId
+ * @requires module:modules/userId
+ */
+
+import * as utils from '../src/utils.js'
+import {submodule} from '../src/hook.js'
+
+const MODULE_NAME = 'originFloc';
+
+/**
+ * Add meta tag to support enabling of floc origin trial
+ * @function
+ * @param {string} token - configured token for origin-trial
+ */
+function enableOriginTrial(token) {
+  const tokenElement = document.createElement('meta')
+  tokenElement.httpEquiv = 'origin-trial'
+  tokenElement.content = token
+  document.head.appendChild(tokenElement)
+}
+
+/**
+ * Get the interest cohort.
+ * @param successCallback
+ * @param errorCallback
+ */
+function getFlocData(successCallback, errorCallback) {
+  document.interestCohort()
+    .then((data) => {
+      successCallback(data);
+    }).catch((error) => {
+      errorCallback(error);
+    });
+}
+
+/**
+ * Encode the id
+ * @param value
+ * @returns {string|*}
+ */
+function encodeId(value) {
+  const result = {};
+  if (value) {
+    result.flocId = value;
+    utils.logInfo('Decoded value ' + JSON.stringify(result));
+    return result;
+  }
+  return undefined;
+}
+
+/** @type {Submodule} */
+export const flocIdSubmodule = {
+  /**
+   * used to link submodule with config
+   * @type {string}
+   */
+  name: MODULE_NAME,
+
+  /**
+   * decode the stored id value for passing to bid requests
+   * @function
+   * @param {string} value
+   * @returns {{flocId:{ id: string }} or undefined if value doesn't exists
+   */
+  decode(value) {
+    return (value) ? encodeId(value) : undefined;
+  },
+  /**
+   * If chrome and cohort enabled performs action to obtain id and return a value in the callback's response argument
+   * @function
+   * @param {SubmoduleConfig} [config]
+   * @returns {IdResponse|undefined}
+   */
+  getId(config) {
+    const isChrome = !!window.chrome && (!!window.chrome.webstore || !!window.chrome.runtime);
+
```

Comment:
@smenzer @patmmccann For checking the chrome, should window.chrome check suffice? I was unable to test window.chrome.webstore || !!window.chrome.runtime as webstore/runtime is not available

Reply:
i'm not sure...sorry!